src/test/java/uk/co/ramp/people/PopulationGeneratorTest.java

```diff
@@ -0,0 +1,264 @@
+package uk.co.ramp.people;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import uk.co.ramp.io.PopulationProperties;
+import uk.co.ramp.io.StandardProperties;
+import uk.co.ramp.utilities.MinMax;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import static org.mockito.Mockito.when;
+import static uk.co.ramp.people.VirusStatus.*;
+
+
+public class PopulationGeneratorTest {
+
+    private static final double DELTA = 1e-6;
+    private PopulationGenerator populationGenerator;
+
+    @Before
+    public void setup() {
+        PopulationProperties populationProperties = Mockito.mock(PopulationProperties.class);
+        StandardProperties properties = Mockito.mock(StandardProperties.class);
+        populationGenerator = new PopulationGenerator(properties, populationProperties);
+    }
+
+    @Test
+    public void getSEIRCounts() {
+
+        Random rand = new Random();
+        int s = rand.nextInt(100);
+        int e = rand.nextInt(100);
+        int i = rand.nextInt(100);
+        int r = rand.nextInt(100);
+
+        Map<Integer, Person> var = new HashMap<>();
+        int start = 0;
+        int end = s;
+        VirusStatus v = SUSCEPTIBLE;
+        createPeople(var, start, end, v);
+
+        start += s;
+        end += e;
+        v = EXPOSED;
+        createPeople(var, start, end, v);
+
+        start += e;
+        end += i;
+        v = INFECTED;
+        createPeople(var, start, end, v);
+
+        start += i;
+        end += r;
+        v = RECOVERED;
+        createPeople(var, start, end, v);
+
+        Map<VirusStatus, Integer> result = PopulationGenerator.getSEIRCounts(var);
+
+        Assert.assertEquals(s, result.get(SUSCEPTIBLE).intValue());
+        Assert.assertEquals(e, result.get(EXPOSED).intValue());
+        Assert.assertEquals(i, result.get(INFECTED).intValue());
+        Assert.assertEquals(r, result.get(RECOVERED).intValue());
+
+    }
+
+    //        PopulationGenerator.createCumulative();
+    @Test
+    public void testCreateCumulative() {
+
```

Comment:
This test is hard to follow, and I looked at the createCumulative method, and it also looks a bit strange.  I think the createCumulative method and its test would be good things to talk through sometime.

Reply:
Yeah, i think the blood/caffeine balance had shifted too far in favour of blood. I'll add comments. 
Effectively, the population data I got was not in a cumulative form, but I didn't want to change the raw data so I am creating a cumulative set on the fly. Some of this stuff can probably be removed in the future anyway.